homeassistant/helpers/template.py

```diff
@@ -382,10 +383,10 @@ def render(
         If limited is True, the template is not allowed to access any function or filter depending on hass or the state machine.
         """
         if self.is_static:
-            if not parse_result or self.hass.config.legacy_templates:
+            if not parse_result or self.hass and self.hass.config.legacy_templates:
                 return self.template
             return self._parse_result(self.template)
-
+        assert self.hass is not None
```

Comment:
If we can't remove the assert, we should add a message:
```suggestion
        assert self.hass is not None, "hass variable not set on template"
```